docs/BestPractices.md

```diff
@@ -0,0 +1,1820 @@
+actual.Should().NotBeEquivalentTo(expected);
+</pre></td>
+</tr><tr>
+<td>
+CollectionAssert.AreNotEquivalent failed. Both collections contain the same elements.
+</td>
+<td>
+Expected collection {SomeProperty: 1, OtherProperty: item, SomeProperty: 2, OtherProperty: other} not be equivalent with collection {SomeProperty: 1, OtherProperty: item, SomeProperty: 2, OtherProperty: other}.
+</td>
+</tr></table>
+
+<table><tr>
+<td><pre lang="csharp">
+CollectionAssert.Contains(actual, expected);
+</pre></td>
+<td><pre lang="csharp">
+actual.Should().Contain(expected);
+</pre></td>
+</tr><tr>
+<td>
+CollectionAssert.Contains failed.
+</td>
+<td>
+Expected collection {SomeProperty: 2, OtherProperty: other} to contain SomeProperty: 1, OtherProperty: item.
+</td>
+</tr></table>
+
+<table><tr>
+<td><pre lang="csharp">
+CollectionAssert.DoesNotContain(actual, expected);
+</pre></td>
+<td><pre lang="csharp">
+actual.Should().NotContain(expected);
+</pre></td>
+</tr><tr>
+<td>
+CollectionAssert.DoesNotContain failed.
+</td>
+<td>
+Expected collection {SomeProperty: 1, OtherProperty: item} to not contain SomeProperty: 1, OtherProperty: item.
+</td>
+</tr></table>
+
+<table><tr>
+<td><pre lang="csharp">
+CollectionAssert.IsSubsetOf(actual, expectedSuperset);
+</pre></td>
+<td><pre lang="csharp">
+actual.Should().BeSubsetOf(expectedSuperset);
+</pre></td>
+</tr><tr>
+<td>
+CollectionAssert.IsSubsetOf failed.
+</td>
+<td>
+Expected collection to be a subset of {SomeProperty: 1, OtherProperty: item}, but items {SomeProperty: 2, OtherProperty: other} are not part of the superset.
+</td>
+</tr></table>
+
+<table><tr>
+<td><pre lang="csharp">
+CollectionAssert.IsNotSubsetOf(actual, expectedSuperset);
+</pre></td>
+<td><pre lang="csharp">
+actual.Should().NotBeSubsetOf(expectedSuperset);
+</pre></td>
+</tr><tr>
+<td>
+CollectionAssert.IsNotSubsetOf failed.
+</td>
+<td>
+Did not expect collection {SomeProperty: 1, OtherProperty: item} to be a subset of {SomeProperty: 1, OtherProperty: item}.
+</td>
+</tr></table>
+
+<table><tr>
+<td><pre lang="csharp">
+CollectionAssert.AllItemsAreNotNull(actual);
+</pre></td>
+<td><pre lang="csharp">
+actual.Should().NotContainNulls();
+</pre></td>
+</tr><tr>
+<td>
+CollectionAssert.AllItemsAreNotNull failed.
+</td>
+<td>
+Expected collection not to contain nulls, but found one at index 0.
+</td>
+</tr></table>
+
+<table><tr>
+<td><pre lang="csharp">
+CollectionAssert.AllItemsAreInstancesOfType(actual, typeof(T));
+</pre></td>
+<td><pre lang="csharp">
+actual.Should().ContainItemsAssignableTo&lt;T&gt;();
+</pre></td>
+</tr><tr>
+<td>
+CollectionAssert.AllItemsAreInstancesOfType failed. Element at index 0 is not of expected type. Expected type:<UnitTests2.MyIdenticalClass>. Actual type:<UnitTests2.MyClass>.
+</td>
+<td>
+Expected collection to contain only items of type UnitTests2.MyIdenticalClass, but item SomeProperty: 1, OtherProperty: item at index 0 is of type UnitTests2.MyClass.
+</td>
+</tr></table>
+
+
+## [ExpectedException]
```

Comment:
Might be worth adding the Assert.ThrowsException equivalent as well, that is if this also applies for MSTest V2.

Reply:
Didn't even know MSTest V2 had that assertion.
I think it should be included.
It also great advertisement why to upgrade to MSTest V2

In fact, this is my favorite example, as the MSTest V1 way is quite ugly.

Reply:
Yes it is. Here is the documentation for MSTest V2: https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting. 
Just putting this out here as well: https://github.com/Microsoft/testfx-docs/blob/master/RFCs/002-Framework-Extensibility-Custom-Assertions.md